src/porepy/numerics/time_step_control.py

```diff
@@ -1,130 +1,191 @@
+            the constant time step.
+        constant_dt: Whether to treat the time step as constant or not. If True, then
+            the time-stepping control algorithm is effectively bypassed. To be
+            precise, the algorithm won't adapt the time step in any situation.
+            Nevertheless, the attributes (such as scheduled times) will still be accesible.
+        dt_min_max: Minimum and maximum permissible time steps. If None, then the
+            minimum time step is set to 0.1% of the final simulation time and the
+            maximum time step is set to 10% of the final simulation time. If given, then
+            the first and second elements of the tuple corresponds to the minimum and
+            maximum time steps, respectively.
+        iter_max: Maximum number of iterations.
+        iter_optimal_range: Optimal iteration range. The first and second elements of the
+            tuple corresponds to the lower and upper bounds of the optimal iteration range.
+        iter_lowupp_factor: Lower and upper multiplication factors. The first and second
+            elements of the tuple corresponds to the lower and upper multiplication
+            factors, respectively. We require the lower multiplication factor to be
+            strictly greater than one, whereas the upper multiplication factor is
+            required to be strictly less than one.
+        recomp_factor: Failed-to-converge recomputation factor. Factor by which the
+            time step will be multiplied in case the solution must be recomputed (see
+            documentation of `next_time_step` method).
+        recomp_max: Failed-to-converge maximum recomputation attempts. The maximum
+            allowable number of consecutive recomputation attempts. If `recomp_max` is
+            exceeded, an error will be raised.
+        print_info. Print time-stepping information.
+
+    Example:
+        # The following is an example on how to construct a time-stepping object
+        tsc = pp.TimeSteppingControl(
+            schedule=[0, 10],
+            dt_init=0.5,
+            dt_min_max=(0.1, 2),
+            iter_max=10,
+            iter_optimal_range=(4, 7),
+            iter_lowupp_factor=(1.1, 0.9),
+            recomp_factor=0.5,
+            recomp_max=5,
+            print_info=True
+        )
+        # To inspect the current attributes of the object
+        print(tsc)
+
+    Attributes:
+        dt (float): Time step.
+        dt_init (float): Intial time step.
+        dt_max (float): Maximum time step.
+        dt_min (float): Minimum time step.
+        is_constant (bool): Constant time step flag.
+        iter_low (int): Lower bound of the optimal iteration range.
+        iter_low_factor (float): Lower multiplication factor. Strictly greater than one.
+        iter_max (int): Maximum number of iterations.
+        iter_upp (int): Upper bound of the optimal iteration range.
```

Comment:
A lower factor that is >1 and an upper factor < 1 is confusing. Is it meaningful to write out what the update formula is somewhere?

Reply:
I agree with @jhabriel  suggestion. Nevertheless, it is necessary to write the actual time step updating formula. For clarity, I suggest splitting next_time_step into simple functions.

Reply:
I'm fine with the suggested naming.

Reply:
Ok. I will rename the parameter and the attributes. I agree with your suggestion Omar, the `next_time_step()` method got too long and has to be split in smaller parts.

Reply:
Omar: I've added a TODO for your suggestion of splitting the next_time_step method. Since I would like to prioritize other aspects of the PR right now. With that, I will resolve this conversation.